rust/capture/src/v0_endpoint.rs

```diff
@@ -300,14 +308,24 @@ pub fn process_single_event(
 #[instrument(skip_all, fields(events = events.len()))]
 pub async fn process_events<'a>(
     sink: Arc<dyn sinks::Event + Send + Sync>,
+    dropper: Arc<TokenDropper>,
     events: &'a [RawEvent],
     context: &'a ProcessingContext,
 ) -> Result<(), CaptureError> {
-    let events: Vec<ProcessedEvent> = events
+    let mut events: Vec<ProcessedEvent> = events
         .iter()
         .map(|e| process_single_event(e, context))
         .collect::<Result<Vec<ProcessedEvent>, CaptureError>>()?;
 
+    events.retain(|e| {
```

Comment:
Feels like I would want a test on the more e2e side of this, not just the unit test as this is a little scary.

Reply:
Done